kiwi/lib/search.py

```diff
@@ -0,0 +1,655 @@
+                'hter'
+            ] = hter
+            search_values['hter'] = hter
+        # Suggest whether to include the word level objective
+        if self.config.options.search_word_level:
+            assert (
+                base_config_dict['data']['train']['output']['target_tags'] is not None
+            )
+            assert (
+                base_config_dict['data']['valid']['output']['target_tags'] is not None
+            )
+            word_level = trial.suggest_categorical('word_level', [True, False])
+            base_config_dict['system']['model']['outputs']['word_level'][
+                'target'
+            ] = word_level
+            base_config_dict['system']['model']['outputs']['word_level'][
+                'gaps'
+            ] = word_level
+            search_values['word_level'] = word_level
+
+        # We search for `sentence_loss_weight` when there is both a word level
+        #   objective and a sentence level objective. Otherwise it does not matter
+        #   to weigh the loss of one objective: for the loss it's the ratio between
+        #   the two objectives that matters, not the absolute value of one loss
+        #   separately.
+        word_level_config = base_config_dict['system']['model']['outputs']['word_level']
+        sentence_level_config = base_config_dict['system']['model']['outputs'][
+            'sentence_level'
+        ]
+        if (
+            word_level_config.get('target')
+            and sentence_level_config.get('hter')
+            and self.config.options.sentence_loss_weight
+        ):
+            sentence_loss_weight = get_suggestion(
+                trial, 'sentence_loss_weight', self.config.options.sentence_loss_weight
+            )
+            base_config_dict['system']['model']['outputs'][
+                'sentence_loss_weight'
+            ] = sentence_loss_weight
+            search_values['sentence_loss_weight'] = sentence_loss_weight
+
+        # The same logic applies here: when search whether to include the sentence level
+        #   objective (`search_hter`) then we will only want to search the
+        #   `sentence_loss_weight` when we are actually including the sentence level
+        #   objevtive (`hter = True`). See the above definition of the variable `hter`.
+        if (
+            word_level_config.get('target')
+            and self.config.options.search_hter
+            and self.config.options.sentence_loss_weight
+        ):
+            if hter:
+                sentence_loss_weight = get_suggestion(
+                    trial,
+                    'sentence_loss_weight',
+                    self.config.options.sentence_loss_weight,
+                )
+                base_config_dict['system']['model']['outputs'][
+                    'sentence_loss_weight'
+                ] = sentence_loss_weight
+                search_values['sentence_loss_weight'] = sentence_loss_weight
```

Comment:
These two conditions are incredibly similar, can't we collapse them into a single If?

Reply:
Yes let me see what I can do